master/buildbot/process/build.py

```diff
@@ -225,19 +227,22 @@ def setupOwnProperties(self):
             props.setProperty("codebase", source.codebase, "Build")
             props.setProperty("project", source.project, "Build")
 
-    def setupWorkerForBuilder(self, workerforbuilder):
```

Comment:
Previously setupWorkerForBuilder did 2 things:
 - Figured out the worker properties to add to build properties
 - Assigned the worker to the build

Since the BuildRequestDistributor also needs worker properties to figure out whether a build can be started for specific build request, this method was split into two so that it's possible to retrieve worker properties without creating a build.

We could probably merge this method into setupProperties, because there's no reason for these methods to be separate.